src/Compilers/CSharp/Portable/Symbols/Source/SourceNamedTypeSymbol_Bases.cs

```diff
@@ -319,6 +319,12 @@ internal override ImmutableArray<NamedTypeSymbol> GetDeclaredInterfaces(ConsList
                 }
             }
 
+            if (declaration.Kind == DeclarationKind.Record)
+            {
+                var type = DeclaringCompilation.GetWellKnownType(WellKnownType.System_IEquatable_T);
+                baseInterfaces.Add(type.Construct(this));
```

Comment:
Does this cause us to implement the interface "non-nullably?" Should we instead implement the interface with a nullable type argument, e.g. `record Rec : IEquatable<Rec?>`

Reply:
I do not see mention of IEquatable in the [records spec](https://github.com/dotnet/csharplang/blob/master/proposals/records.md).

Reply:
The parameter to `IEquatable<T>.Equals()` is `[AllowNull] T other`.

Reply:
> I do not see mention of IEquatable in the records spec.

Good catch, thanks. I've updated https://github.com/dotnet/csharplang/pull/3552.

Reply:
> The parameter to IEquatable<T>.Equals() is [AllowNull] T other.

So it is, thank you! It might be good to add tests that demonstrate the synthesized "strongly typed Equals" methods accept null arguments without warnings or runtime errors.